### PR TITLE
Ios objectivec

### PR DESCRIPTION
**Pull request by midfords**

### CODE REVIEW

File: Lunchr/Lunchr/HttpUtils.m [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+//
+//  Created by Sean-MacbookPro on 2/26/15.
+//
+//
+
+#import "HttpUtils.h"
+
+@implementation HttpUtils
+
++ (bool)registerUser:(NSString *)email
```
Comment (paymog) [inline Lunchr/Lunchr/HttpUtils.m]: why is this `registerUser` as opposed to `toEmail`?

File: Lunchr/Lunchr/HttpUtils.m [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    return true;
+}
+
++ (bool)loginUser:(NSString *)email
+       toPassword:(NSString *)password {
+    
+    @try {
+        
+        NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"http://54.200.225.219/api/users/authenticate"]];
+        //NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"http://localhost:3000/api/users/authenticate"]];
```
Comment (paymog) [inline Lunchr/Lunchr/HttpUtils.m]: Remove comment

File: Lunchr/Lunchr/HttpUtils.m [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                                        nil];
+        NSError *error = nil;
+        
+        NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonDictionary options:NSJSONWritingPrettyPrinted error:&error];
+        NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+        NSUInteger postLength = [jsonString length];
+        
+        [request setHTTPMethod:@"POST"];
+        [request setValue:@"application/json" forHTTPHeaderField:@"Content-type"];
+        [request setValue:[NSString stringWithFormat:@"%d", (int)postLength] forHTTPHeaderField:@"Content-length"];
```
Comment (paymog) [inline Lunchr/Lunchr/HttpUtils.m]: seems like you're repeating a lot of code here like the request above.

File: Lunchr/Lunchr/LoginViewController.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+//
+
+#import <UIKit/UIKit.h>
+
+@interface LoginViewController : UIViewController
+
+- (IBAction)loginButtonTapped:(UIButton *)sender;
+
+@property (strong, nonatomic) IBOutlet UITextField *emailField;
+@property (strong, nonatomic) IBOutlet UITextField *passField;
```
Comment (paymog) [inline Lunchr/Lunchr/LoginViewController.h]: passwordField

File: Lunchr/Lunchr/LoginViewController.m [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+//
+//
+
+#import "LoginViewController.h"
+#import "HttpUtils.h"
+
+@implementation LoginViewController
+
+@synthesize emailField;
+@synthesize passField;
```
Comment (paymog) [inline Lunchr/Lunchr/LoginViewController.m]: passwordField

File: Lunchr/Lunchr/RegisterViewController.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+//
+//  Created by Sean-MacbookPro on 2/24/15.
+//
+//
+
+#import <UIKit/UIKit.h>
+
+@interface RegisterViewController : UIViewController
+
+@property (strong, nonatomic) IBOutlet UITextField *firstField;
```
Comment (paymog) [inline Lunchr/Lunchr/RegisterViewController.h]: firstNameField

File: Lunchr/Lunchr/RegisterViewController.h [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+@property (strong, nonatomic) IBOutlet UITextField *lastField;
```
Comment (paymog) [inline Lunchr/Lunchr/RegisterViewController.h]: lastNameField

File: Lunchr/Lunchr/RegisterViewController.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+@property (strong, nonatomic) IBOutlet UITextField *emailField;
+@property (strong, nonatomic) IBOutlet UITextField *passField;
```
Comment (paymog) [inline Lunchr/Lunchr/RegisterViewController.h]: passwordField

File: Lunchr/Lunchr/RegisterViewController.m [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+@interface RegisterViewController ()
+
+@end
+
+@implementation RegisterViewController
+
+@synthesize firstField;
+@synthesize lastField;
+@synthesize emailField;
```
Comment (paymog) [inline Lunchr/Lunchr/RegisterViewController.m:20]: see comments above

File: Lunchr/Lunchr.xcodeproj/xcuserdata/sean.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 10]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kjrcda) [inline Lunchr/Lunchr.xcodeproj/xcuserdata/sean.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist]: Will this file effect anything while running the app?

**Comment by kjrcda** [comment]
Good work

File: Lunchr/Lunchr/HttpUtils.m (hunk 3 above)
Comment (midfords) [inline Lunchr/Lunchr/HttpUtils.m]: Removed the duplicate code and moved it to new method "postRequest(json, url)".

File: Lunchr/Lunchr/HttpUtils.m [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    
+    NSDictionary *jsonDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
+                                    email, @"email",
+                                    password, @"password",
+                                    firstName, @"firstname",
+                                    lastName, @"lastname",
+                                    nil];
+    NSString *url = @"http://54.200.225.219/api/users/register";
+    
+    return [HttpUtils postRequest:jsonDictionary toURL:url];
```
Comment (kjrcda) [inline Lunchr/Lunchr/HttpUtils.m:63]: this will return a Boolean value correct?

Comment (midfords) [inline Lunchr/Lunchr/HttpUtils.m:63]: Yes.

**Comment by paymog** [comment]
Ship it!

**Pull request merged**